examples/bandtrade_ksilo.py

```diff
@@ -0,0 +1,382 @@
+                and today_open > yesterday_close
+                and current_price > today_lower_band
+            ):
+                yesterday_upper_band = bband[0][-2]
+                if current_price > yesterday_upper_band:
+                    return {}
+
+                buy_indicators[symbol] = {
+                    "lower_band": bband[2][-2:].tolist(),
+                }
+                shares_to_buy = await self.calc_qty(
+                    current_price,
+                    trade_fee_precentage,
+                )
+                tlog(
+                    f"[{self.name}][{now}] Submitting buy for {shares_to_buy} shares of {symbol} at {current_price}"
+                )
+                tlog(f"indicators:{buy_indicators[symbol]}")
+                actions[symbol] = {
+                    "side": "buy",
+                    "qty": str(shares_to_buy),
+                    "type": "limit",
+                    "limit_price": str(current_price),
+                }
+
+        return actions
+
+    async def handle_sell_side(
+        self,
+        symbols_position: Dict[str, float],
+        data_loader: DataLoader,
+        now: datetime,
+        trade_fee_precentage: float,
+    ) -> Dict[str, Dict]:
+        actions = {}
+
+        for symbol, position in symbols_position.items():
+            if position == 0:
+                continue
+
+            current_price = data_loader[symbol].close[now]
+            resampled_close = self.calc_close(symbol, data_loader, now)
+            bband = BBANDS(
+                resampled_close,
+                timeperiod=7,
+                nbdevdn=1,
+                nbdevup=1,
+                matype=MA_Type.EMA,
+            )
+
+            yesterday_upper_band = bband[0][-2]
+
+            print(
+                f"\ncurrent_price > yesterday_upper_band : {current_price > yesterday_upper_band}({current_price} < {yesterday_upper_band})"
+            )
+
+            if current_price > yesterday_upper_band:
+                sell_indicators[symbol] = {
+                    "upper_band": bband[0][-2:].tolist(),
+                    "lower_band": bband[2][-2:].tolist(),
+                }
+
+                tlog(
+                    f"[{self.name}][{now}] Submitting sell for {position} shares of {symbol} at market"
+                )
+                tlog(f"indicators:{sell_indicators[symbol]}")
+                actions[symbol] = {
+                    "side": "sell",
+                    "qty": str(position),
+                    "type": "limit",
+                    "limit_price": str(current_price),
+                }
+
+        return actions
+
+    async def should_run_all(self):
+        return False
+    
+    async def run(
+        self,
+        symbol: str,
+        now: datetime,
+        position: float = None,
+        portfolio_value: float = None,
+        debug: bool = False,
+        backtesting: bool = False,
+        minute_history: pd.DataFrame = None,
```

Comment:
@ksilo, I wonder if you shouldn't move additional variables to the context, including symbol(s), position(s), minute_history. Maybe the run function should get controlling parameters such as debug level, timestamp, etc.